src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -323,6 +323,10 @@ internal MetadataContextItem(TMetadataContext metadataContext)
             {
                 this.MetadataContext = metadataContext;
             }
+
+            // This override is needed to keep the DkmDataItem from being unloaded ahead of this component.
```

Comment:
Perhaps a better comment might be: This override is needed to prevent the Concord dispatcher from unloading components under the EE when code from an EE component is currently running.

Reply:
Thank you!

---
In reply to: [143874341](https://github.com/dotnet/roslyn/pull/22644#discussion_r143874341) [](ancestors = 143874341)